ngraph_bridge/ngraph_encapsulate_impl.cc

```diff
@@ -0,0 +1,445 @@
+/*******************************************************************************
+ * Copyright 2019 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+#include <cstdlib>
+#include <mutex>
+#include <utility>
+
+#include "tensorflow/core/common_runtime/dma_helper.h"
+#include "tensorflow/core/common_runtime/function.h"
+#include "tensorflow/core/common_runtime/optimization_registry.h"
+#include "tensorflow/core/framework/graph.pb.h"
+#include "tensorflow/core/framework/node_def_util.h"
+#include "tensorflow/core/framework/op.h"
+#include "tensorflow/core/framework/op_kernel.h"
+#include "tensorflow/core/framework/tensor.h"
+#include "tensorflow/core/graph/graph.h"
+#include "tensorflow/core/graph/graph_constructor.h"
+
+#include "ngraph/event_tracing.hpp"
+#include "ngraph/runtime/backend.hpp"
+
+#if defined NGRAPH_DISTRIBUTED
+#include "ngraph/distributed.hpp"
+#endif
+
+#include "logging/ngraph_log.h"
+#include "ngraph_bridge/ngraph_backend_manager.h"
+#include "ngraph_bridge/ngraph_builder.h"
+#include "ngraph_bridge/ngraph_cluster_manager.h"
+#include "ngraph_bridge/ngraph_encapsulate_impl.h"
+#include "ngraph_bridge/ngraph_encapsulate_op.h"
+#include "ngraph_bridge/ngraph_mark_for_clustering.h"
+#include "ngraph_bridge/ngraph_timer.h"
+#include "ngraph_bridge/ngraph_utils.h"
+
+#if defined(NGRAPH_TF_ENABLE_VARIABLES_AND_OPTIMIZERS)
+#include "ngraph_bridge/enable_variable_ops/ngraph_catalog.h"
+#include "ngraph_bridge/enable_variable_ops/ngraph_var.h"
+#endif
+
+using namespace std;
+namespace ng = ngraph;
+
+namespace tensorflow {
+
+namespace ngraph_bridge {
+
+// Ngraph Encapsulate Implementation class for EncapsulateOp class
+//---------------------------------------------------------------------------
+//  NGraphEncapsulateImpl::ctor
+//---------------------------------------------------------------------------
+NGraphEncapsulateImpl::NGraphEncapsulateImpl()
+    : m_graph(OpRegistry::Global()), m_freshness_tracker(nullptr) {
+  my_instance_id = s_instance_count;
+  s_instance_count++;
+}
+
+// Get tensorflow input tensors, input shapes, static_inputs to Compute
+// Signature
```

Comment:
The comment says it needs 3 things and it computes signature from that. This function rather has a single input tf_input_tensors and outputs 3 things: input_shapes, static_input_map, signature_ss

The function is less `ComputeSignature` and more `ComputeSignatureInputShapeAndStaticInputs` (its unwieldily though)

Reply:
Will update the comment :)